dataland-backend/src/main/kotlin/org/dataland/datalandbackend/controller/DataController.kt

```diff
@@ -2,46 +2,43 @@ package org.dataland.datalandbackend.controller
 import org.dataland.datalandbackend.api.DataAPI
-import org.dataland.datalandbackend.interfaces.DataStoreInterface
-import org.dataland.datalandbackend.model.CompanyAssociatedDataSet
-import org.dataland.datalandbackend.model.DataIdentifier
-import org.dataland.datalandbackend.model.DataSetMetaInformation
+import org.dataland.datalandbackend.interfaces.DataManagerInterface
+import org.dataland.datalandbackend.model.CompanyAssociatedData
+import org.dataland.datalandbackend.model.DataManagerInputToGetData
+import org.dataland.datalandbackend.model.DataMetaInformation
 import org.dataland.datalandbackend.model.StorableDataSet
 import org.springframework.http.ResponseEntity
 
 /**
  * Implementation of the API for data exchange
- * @param dataStore implementation of the DataStoreInterface that defines how uploaded data is to be stored
- */
+ * @param dataManager implementation of the DataManagerInterface that defines how
+ * Dataland handles data */
 
 abstract class DataController<T>(
-    var dataStore: DataStoreInterface,
+    var dataManager: DataManagerInterface,
     var objectMapper: ObjectMapper,
     val clazz: Class<T>
 ) : DataAPI<T> {
     private val dataType = clazz.toString().substringAfterLast(".")
 
-    override fun getData(): ResponseEntity<List<DataSetMetaInformation>> {
-        return ResponseEntity.ok(this.dataStore.listDataSets())
-    }
-
-    override fun postData(companyAssociatedDataSet: CompanyAssociatedDataSet<T>): ResponseEntity<String> {
-        return ResponseEntity.ok(
-            this.dataStore.addDataSet(
-                StorableDataSet(
-                    companyId = companyAssociatedDataSet.companyId,
-                    dataType = dataType,
-                    data = objectMapper.writeValueAsString(companyAssociatedDataSet.dataSet)
-                )
+    override fun postCompanyAssociatedData(companyAssociatedData: CompanyAssociatedData<T>):
+        ResponseEntity<DataMetaInformation> {
+        val dataIdOfPostedData = dataManager.addDataSet(
+            StorableDataSet(
+                companyAssociatedData.companyId, dataType,
+                data = objectMapper.writeValueAsString(companyAssociatedData.data)
             )
         )
+        return ResponseEntity.ok(
+            DataMetaInformation(dataIdOfPostedData, dataType, companyAssociatedData.companyId)
+        )
     }
 
-    override fun getCompanyAssociatedDataSet(dataId: String): ResponseEntity<CompanyAssociatedDataSet<T>> {
-        val dataset = this.dataStore.getStorableDataSet(DataIdentifier(dataId = dataId, dataType = dataType))
+    override fun getCompanyAssociatedDataSet(dataId: String): ResponseEntity<CompanyAssociatedData<T>> {
         return ResponseEntity.ok(
-            CompanyAssociatedDataSet(
-                objectMapper.readValue(dataset.data, clazz), dataset.companyId
+            CompanyAssociatedData(
+                companyId = dataManager.searchDataMetaInfo(dataId).first().companyId,
```

Comment:
wenn hier sichergestellt sein muss, dass genau ein element zurück kommt, sollten wir das so abbilden. 
Das heißt: wir sollten searchDataMetaInfo(dataId) auch nur ein Element zurückliefern lassen. Wenn es da intern mehr als ein Element findet, soll ein Fehler fliegen.

Reply:
Wenn eine dataId als Input gegeben wird (dataId also nicht leer ist), dann kann `searchDataMetaInfo(dataId)` per Code nur eine Liste mit einem einzigen `CompanyAssociatedDataSet`-Objekt zurückgeben, weil das aus einer Map geholt wird, die ja per Definition keine doppelten Keys zulässt:

![image](https://user-images.githubusercontent.com/99255296/161242879-e3608629-5e17-439c-b4be-52d0b262aaf2.png)

Die Liste, die hier returnt wird, kann also nur ein Element haben.

Daher ist das `first()` eigentlich nur ein Mittel zum  Zweck, um das Element aus der Liste zu holen.

Was sollen wir also tun? Ich könnte im DataController trotzdem einen Check einführen, der bei mehr als einem Element eine Exception wirft. Passt das?

Reply:
Die Exception würde aber bei aktuellem Stand des Codes aber in keinem Szenario eintreten.  Wäre also toter Code.

Reply:
Bitte wie besprochen ändern

Reply:
ÜBernheme ich

Reply:
ok